litejpeg/core/huffman/huffmancore.py

```diff
@@ -0,0 +1,479 @@
+
+                  ).Elif(hfw_running & (
+                         (num_fifo_wrs == 0) | (fifo_wrt_cnt == num_fifo_wrs)),
+
+                         [If(i+(num_fifo_wrs*8) < width_word,
+                             word_reg[width_word-1-i].eq(
+                                 word_reg[width_word-1-(num_fifo_wrs*8)-i]))
+                             for i in range(width_word)],
+
+                         bit_ptr.eq(bit_ptr - (num_fifo_wrs*8)),
+                         hfw_running.eq(0),
+                         fifo_wrt_cnt.eq(0),
+                         first_rle_word.eq(0),
+                         state.eq(2),
+                         ready_one.eq(1)))
+         # VLI state
+         .Elif(state == 2,
+
+               If(hfw_running == 0,
+
+                  [If(i < vli_ext_size_next_next_next,
+                      word_reg[width_word-1-bit_ptr-i].eq(
+                          vli_ext_next_next_next[
+                              vli_ext_size_next_next_next-1-i]))
+                      for i in range(width_word)],
+                  bit_ptr.eq(bit_ptr + vli_ext_size_next_next_next),
+                  hfw_running.eq(1)
+
+                  ).Elif(hfw_running & (
+                             (num_fifo_wrs == 0) | (
+                                 fifo_wrt_cnt == num_fifo_wrs)),
+
+                         [If(i+(num_fifo_wrs*8) < width_word,
+                             word_reg[width_word-1-i].eq(
+                             word_reg[width_word-1-(num_fifo_wrs*8)-i]))
+                             for i in range(width_word)],
+                         bit_ptr.eq(bit_ptr - (num_fifo_wrs*8)),
+                         hfw_running.eq(0),
+                         fifo_wrt_cnt.eq(0),
+
+                         # Taking care of the last block.
+                         If(self.word_count == 63,
+                            If((bit_ptr - (num_fifo_wrs*8)) != 0,
+                               state.eq(3))
+                            .Else(
+                                  state.eq(0))
+                            ).Else(
+                                   state.eq(1),
+                                   self.ready_data_write.eq(1))))
+         # Padding state
+         .Else(
+               If(hfw_running == 0,
+
+                  [(If(i < bit_ptr,
+                       pad_byte[7-i].eq(word_reg[width_word-1-i])
+                       ).Else(
+                              pad_byte[7-i].eq(1)
+                     )) for i in range(8)],
+                  pad_reg.eq(1),
+                  bit_ptr.eq(8),
+                  hfw_running.eq(1)
+                  ).Elif(
+                      hfw_running &
+                      ((num_fifo_wrs == 0) | (fifo_wrt_cnt == num_fifo_wrs)),
+
+                      bit_ptr.eq(0),
+                      hfw_running.eq(0),
+                      pad_reg.eq(0),
+                      state.eq(0))
+         )]
+
+
+class HuffmanEncoder(PipelinedActor, Module):
+    """
```

Comment:
Some extra newlines in this comment would be good.

Reply:
Done